frontend/src/metabase/common/components/EntityPicker/components/EntityPickerModal/EntityPickerModal.tsx

```diff
@@ -122,53 +160,173 @@ export function EntityPickerModal<
 
   const { open } = useModalOpen();
 
-  const tabModels = useMemo(
-    () => passedTabs.map(t => t.model).filter(Boolean),
-    [passedTabs],
+  const [debouncedSearchQuery, setDebouncedSearchQuery] = useState(searchQuery);
+  useDebounce(() => setDebouncedSearchQuery(searchQuery), 200, [searchQuery]);
+
+  const { data, isFetching } = useSearchQuery(
+    {
+      q: debouncedSearchQuery,
+      models: searchModels,
+      context: "entity-picker",
+      ...searchParams,
+    },
+    {
+      skip: !debouncedSearchQuery || searchScope === "folder",
+    },
   );
 
+  const finalSearchResults = useMemo(() => {
+    if (searchScope === "folder") {
+      if (!scopedSearchResults) {
+        return null;
+      }
+      return searchResultFilter
+        ? searchResultFilter(scopedSearchResults as SearchResult[])
+        : scopedSearchResults;
+    } else {
+      if (isFetching || !data) {
+        return null;
+      }
+      return searchResultFilter ? searchResultFilter(data.data) : data.data;
```

Comment:
nit: `searchResultFilter` could default to an identity function to avoid 2 conditions here